crates/oxc_linter/src/rules/eslint/no_useless_call.rs

```diff
@@ -0,0 +1,207 @@
+use oxc_ast::{
+    ast::{CallExpression, ChainElement, Expression, MemberExpression},
+    match_member_expression, AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::cmp::ContentEq;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_useless_call_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Avoid unnecessary use of .call() and .apply()")
```

Comment:
eslint has 2 different messages for `.call()` and `.apply()`. Would like to see 2 different diagnostics here too :)
https://github.com/eslint/eslint/blob/67dd82ab88d784b6f36e471b6a5c6f64e37f9485/lib/rules/no-useless-call.js#L66

Reply:
I’ve addressed it in the commit below!
https://github.com/oxc-project/oxc/pull/8789/commits/a7c0e97c5151e2fbd69ba0f6d8d8819da37cd1f5